### PR TITLE
1.0.5 release

### PR DESCRIPTION
**Pull request by Rykh**
### Release Note

| Patch key  | Description |
| ------------- | ------------- |
|MDVA-30841|Fixes the issue with layered navigation, where the 'No' value for boolean type product attributes was not included in layered navigation if Elasticsearch was used as a search engine. Magento version 2.3.3|
|MDVA-28191|Fixes the issue where no payment methods are loaded during order creation via Magento Admin panel.|
|MDVA-29959|Fixes the issue where restricted admin user with 'Companies' permissions is not allowed to delete company account.|
|MDVA-30265|Fixes the issue where shipment tracking link stops working after Invoice creation.|
|MDVA-28409|Fixes the issue where the "sales_clean_quotes" cron job fails with out-of-memory error when the number of expired quotes in the database is huge.|
|MDVA-30593|Fixes the issue where quotes, that expired according to the Quote Lifetime setting, are not cleaned up.|
|MDVA-30107|Fixes the issue where store switcher doesn't work as expected if different base URLs are used for store views.|
|MDVA-28763|Fixes the issue where product image is getting duplicated after updating product information using REST API more than once.|
|MDVA-30284|Fixes the issue where Catalog Search indexer fails due to the following Elasticsearch error: `limit of total fields in index has been exceeded`.|
|MDVA-29042|Fixes the issue where Catalog permissions were changed to `Allow` automatically after new product was added to the shared catalog.|
|MDVA-30428|Fixes the issue where customers cannot add a product to wishlist if this product is assigned to a custom inventory source.|
|MDVA-28661|Fixes the issue where an error is thrown in the `Company Users` company account section after company admin is changed.|

### Fixed Issues
1. MDVA-30841
2. MDVA-28191
3. MDVA-29959
4. MDVA-30265
5. MDVA-28409
6. MDVA-30593
7. MDVA-30107
8. MDVA-30623
9. MDVA-28763
10. MDVA-30284
11. MDVA-29042
12. MDVA-30428
13. MDVA-28661

### CODE REVIEW

File: patches.json [hunk 1]
```diff
@@ -225,5 +228,108 @@
...
+  "MDVA-28409": {
+    "magento/magento2-base": {
+      "Fixes the issue where the 'sales_clean_quotes' cron job fails with out-of-memory error when the number of expired quotes in the database is huge.": {
+        ">=2.3.4 <2.3.6 || 2.4.0": {
+          "file": "os/MDVA-28409__cleaning_old_quotes_from_sales_clean_quotes_table__2.3.4.patch"
+        },
+        ">=2.3.0 <2.3.4": {
+          "file": "os/MDVA-30593__cleaning_old_quotes_from_sales_clean_quotes_table__2.3.3.patch"
+        }
+      }
```
Comment (ilima-ebay) [inline patches.json:268]: I do not think MDVA-28409 and MDVA-30593 should be merged together. They are different issues 
Please compare test case between MDVA-28409 (backport of MC-34782) and MDVA-30593 (backport of MC-19798)

File: patches.json [hunk 2]
```diff
@@ -225,5 +228,108 @@
...
+        ">=1.1.5 <1.2.2": {
+          "file": "os/MDVA-30428__fixes_add_a_product_with_custom_inventory_to_wishlist__1.1.5.patch"
+        }
+      }
+    }
+  },
+  "MDVA-28661": {
+    "magento/magento2-b2b-base": {
+      "Fixes the issue where an error is thrown in the `Company Users` company account section after company admin is changed.": {
+        ">=1.1.0 <1.2.2": {
```
Comment (ilima-ebay) [inline patches.json:329]: This patch is not applicable to 1.1.5-p2. Please review version contraints.
Suggested: `>=1.1.0 <1.1.5-p2 || >=1.2.0 <1.2.2`

**Review by ilima-ebay — CHANGES_REQUESTED** [review]
- [ ] Validate that patches.json is structured correctly and the correct composer package is used as well as patch directory 
- [x] Code matches the original core fix from specified ticket
- [x] Check that code was tested by QA in the original ticket
- [x] Check the automated builds are green

File: patches.json (hunk 1 above)
Comment (Rykh) [reply patches.json:268]: Fixed

File: patches.json (hunk 2 above)
Comment (Rykh) [reply patches.json:329]: Fixed

Comment (viktym) [reply patches.json:329]: @ilima-ebay  @Rykh  MQP build shows that the patch is applicable to 1.1.5-p2
I checked manually and it's applicable
We need to check the http://web2.sparta.corp.magento.com/dev/support/tools/patchchk/ tool if it works correctly with security releases

Comment (ilima-ebay) [reply patches.json:329]: @viktym @Rykh yes, I checked the patch checker tool and the 2.3.5-p2 installation was missing b2b that's why it was not applicable. I fixed it and it now shows applicable. @Rykh  i think you can change it back to `>=1.1.0 <1.2.2`. Sorry about that

File: patches.json [hunk 3]
```diff
@@ -225,5 +228,108 @@
...
+        }
+      }
+    },
+    "magento/inventory-metapackage": {
+      "Fixes the issue where customers cannot add a product to wishlist if this product is assigned to a custom inventory source.": {
+        ">=1.1.5 <1.2.2": {
+          "file": "os/MDVA-30428__fixes_add_a_product_with_custom_inventory_to_wishlist__1.1.5.patch"
+        }
+      }
+    }
```
[the remark this answers is not in the document]
Comment (Rykh) [reply patches.json:336]: Fixed

**Review by viktym — APPROVED** [review]

**Review by ilima-ebay — APPROVED** [review]
- [x] Validate that patches.json is structured correctly and the correct composer package is used as well as patch directory 
- [x] Code matches the original core fix from specified ticket
- [x] Check that code was tested by QA in the original ticket
- [x] Check the automated builds are green

Approved. Checklist complete.

**Pull request merged**